Readme.md

```diff
@@ -42,6 +42,78 @@ $ npm install koa
  - [Introduction Screencast](http://knowthen.com/episode-3-koajs-quickstart-guide/) - An introduction to installing and getting started with Koa
 
 ## Example
+```js
+const Koa = require('koa');
+const app = new Koa();
+
+// logger
+
+app.use(function (ctx, next){
+  const start = new Date;
+  next(ctx).then(function() {
+    const ms = new Date - start;
+    console.log(`${ctx.method} ${ctx.url} - ${ms}`);
+  });
+});
+
+// response
+
+app.use(function (ctx){
+  ctx.body = 'Hello World';
+});
+
+app.listen(3000);
+```
+
+## Example with asyncFunction(babel-node required)
+
+```js
+const Koa = require('koa');
+const app = new Koa();
+
+// logger
+
+app.use(async (ctx, next) => {
+  const start = new Date;
+  await next(ctx);
+  const ms = new Date - start;
+  console.log(`${ctx.method} ${ctx.url} - ${ms}`);
+});
+
+// response
+
+app.use(function (ctx){
+  ctx.body = 'Hello World';
+});
+
+app.listen(3000);
+```
+
+## Example with generator
+
+```js
+const Koa = require('koa');
+const app = new Koa();
+
+// logger
+
+app.use(function *(ctx, next){
+  const start = new Date;
+  yield next(ctx);
+  const ms = new Date - start;
+  console.log(`${ctx.method} ${ctx.url} - ${ms}`);
+});
+
+// response
+
+app.use(function *(ctx){
+  ctx.body = 'Hello World';
+});
+
+app.listen(3000);
+```
+
+## Example with old version, Koa < 2
```

Comment:
So , if I understand, I remove the generator function example for new koa version or keep it with a mention saying : it will eventually be removed when async functions are native.?